new: Add translation for Sodium's video settings screen

FlashyReese opened the pull request:
This PR is cherry picked from #100 with additional changes of:
- Replaced all `I18n#translate` with `Text`

FlashyReese committed:
new: Add translation for Sodium's video settings screen

Co-Authored-By: AMereBagatelle <56981737+AMereBagatelle@users.noreply.github.com>

AMereBagatelle reviewed (approved):
Looks good, just make sure all the translations are correct in-game (but I assume you've already done that)

bytzo commented on build.gradle:
```diff
@@ -30,6 +30,10 @@ dependencies {
+
+    // Fabric Resource Loader for translations
+    modCompile "net.fabricmc.fabric-api:fabric-resource-loader-v0:${project.fabric_resource_loader_version}"
+    include "net.fabricmc.fabric-api:fabric-resource-loader-v0:${project.fabric_resource_loader_version}"
```
```suggestion
    modCompile fabricApi.module("fabric-resource-loader-v0", project.fabric_api_version)
    include fabricApi.module("fabric-resource-loader-v0", project.fabric_api_version)
```
Instead of using the hash for the Fabric API Resource Loader, `fabricApi.module` can be used which will automatically fetch the correct hash for the specified Fabric API version. This improves readability in the `gradle.properties` file and simplifies upgrading to newer versions of the Fabric API.

bytzo commented on gradle.properties:
```diff
@@ -12,6 +12,9 @@ mod_version=0.1.1
 archives_base_name=sodium-fabric
 
+# Fabric API Resource Loader version
+fabric_resource_loader_version=0.2.5+059ea86602
```
```suggestion
# Fabric API version
fabric_api_version=0.30.0+1.16
```
See comment in `build.gradle`.

FlashyReese committed:
change: Implement TextProvider for OptionImpact

bytzo commented on src/main/resources/assets/sodium/lang/en_us.json:
```diff
@@ -0,0 +1,57 @@
+  "sodium.options.use_chunk_face_culling.name": "Use Chunk Face Culling",
+  "sodium.options.use_chunk_face_culling.tooltip": "If enabled, an additional culling pass will be performed on the CPU to determine which planes of a chunk mesh are visible. This can eliminate a large number of block faces very early in the rendering process, saving memory bandwidth and time on the GPU.",
+  "sodium.options.use_compact_vertex_format.name": "Use Compact Vertex Format",
+  "sodium.options.use_compace_vertex_format.tooltip": "If enabled, a more compact vertex format will be used for chunk meshes which limits the precision of vertex attributes. This format will reduce graphics memory usage and bandwidth requirements by around 40%, but could cause z-fighting/flickering texture issues in some edge cases.",
```
```suggestion
  "sodium.options.use_compact_vertex_format.tooltip": "If enabled, a more compact vertex format will be used for chunk meshes which limits the precision of vertex attributes. This format will reduce graphics memory usage and bandwidth requirements by around 40%, but could cause z-fighting/flickering texture issues in some edge cases.",
```
Small typo fix.

FlashyReese committed:
change: Remove additional translation key for performance impact

bytzo commented on src/main/java/me/jellysquid/mods/sodium/client/gui/options/OptionImpact.java:
```diff
@@ -1,13 +1,15 @@
-    HIGH(Formatting.GOLD, "High"),
-    EXTREME(Formatting.RED, "Extreme"),
-    VARIES(Formatting.WHITE, "Varies");
+public enum OptionImpact implements TextProvider{
```
```suggestion
public enum OptionImpact implements TextProvider {
```

bytzo commented on src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumOptionsGUI.java:
```diff
@@ -198,7 +197,7 @@ private void renderOptionTooltip(MatrixStack matrixStack, ControlElement<?> elem
 
         if (impact != null) {
-            tooltip.add(Language.getInstance().reorder(new LiteralText(Formatting.GRAY + "Performance Impact: " + impact.toDisplayString())));
+            tooltip.add(Language.getInstance().reorder(new TranslatableText("sodium.options.performance_impact_string", "", impact.getText()).formatted(Formatting.GRAY)));
```
```suggestion
            tooltip.add(Language.getInstance().reorder(new TranslatableText("sodium.options.performance_impact_string", impact.getText()).formatted(Formatting.GRAY)));
```
I think this empty string can be removed, let me know if I am wrong.

FlashyReese replied:
It can be removed, I just kept it since I didn't want to touch the translation strings.

FlashyReese replied:
Although this should be update in OneSky, further discussion might be required on how Jelly plans to handle.

bytzo replied:
It looks like this may have already been addressed in [`f8c3a67`](https://github.com/CaffeineMC/sodium-fabric/commit/f8c3a67822ca594ee934d5da0fd12097f4daf5d1) (made after the original PR was merged into dev), as the empty string has been removed there. The translation keys on OneSky also seem to have the empty string removed, so it should be safe to remove here.

FlashyReese replied:
Nice catch, I will commit the changes.

bytzo commented on src/main/resources/assets/sodium/lang/en_us.json:
```diff
@@ -0,0 +1,56 @@
+  "sodium.options.use_memory_intrinsics.tooltip": "If enabled, special intrinsics will be used to speed up the copying of client memory in certain vertex-limited scenarios, such as particle and text rendering. This option only exists for debugging purposes and should be left enabled unless you know what you are doing.",
+  "sodium.options.disable_driver_blacklist.name": "Disable Driver Blacklist",
+  "sodium.options.disable_driver_blacklist.tooltip": "If selected, Sodium will ignore the built-in driver blacklist and enable options which are known to be broken with your system configuration. This might cause serious problems and should not be used unless you really do know better. The settings screen must be saved, closed, and re-opened after changing this option in order to reveal previousl ...
+  "sodium.options.performance_impact_string": "%sPerformance Impact: %s",
```
```suggestion
  "sodium.options.performance_impact_string": "Performance Impact: %s",
```
See comment in `SodiumOptionsGUI.java`.

bytzo reviewed (commented):
I've suggested a few small changes, otherwise it looks good to me!

FlashyReese committed:
Apply suggestions from code review

Co-authored-by: Bytzo <bytzo@bytzo.net>

FlashyReese committed:
change: Correct a translation string and formatting with video options screen

Co-authored-by: Bytzo <bytzo@bytzo.net>

i509VCB commented on src/main/java/me/jellysquid/mods/sodium/client/gui/options/OptionImpact.java:
```diff
@@ -17,7 +19,8 @@
-        return this.color + this.text;
+    @Override
+    public Text getText() {
+        return new TranslatableText(this.text).formatted(this.color);
```
Might help with reducing allocations if you construct the text in constructor?

FlashyReese replied:
I'm not sure if this is the case, as Sodium's options are built on Screen initialization. I may be wrong since I'm unsure.
Edit: You are right, it should help reduce allocations

FlashyReese committed:
change: Construct Text for TextProvider within constructor to reduce allocations

FlashyReese committed:
new: Add translations from OneSky

The es_la isn't valid for Minecraft so multiple copies of it was made for all other Spanish speaking countries

FlashyReese committed:
Merge remote-tracking branch 'upstream/1.16.x/next' into translations

FlashyReese committed:
Merge branch '1.16.x/next' into translations

FlashyReese committed:
fix: Resolve conflicts

jellysquid3 commented:
This PR is quite old, and Sodium has since adopted translatable strings in the video settings with the release of Sodium 0.3.
